scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -471,7 +471,18 @@ trait TypedPipe[+T] extends Serializable {
    * Reasonably common shortcut for cases of total associative/commutative reduction
    * returns a ValuePipe with only one element if there is any input, otherwise EmptyValue.
    */
-  def sum[U >: T](implicit plus: Semigroup[U]): ValuePipe[U] = ComputedValue(groupAll.sum[U].values)
+  def sum[U >: T](implicit plus: Semigroup[U]): ValuePipe[U] = {
+    // every 1000 items, compact.
+    lazy implicit val batchedSG = Batched.compactingSemigroup[U](1000)
+    ComputedValue(map { t => ((), Batched[U](t)) }
+      .sumByLocalKeys
+      // remove the Batched before going to the reducers
+      .map { case (_, batched) => batched.sum }
```

Comment:
adding a method to a trait is not binary compatible (in scala 2.11, in scala 2.12 it will be).

do we really think it is worth adding this complexity given the default for `.sum` already keeps 100k items? Why are we suddenly so conservative?

I don't want to add a second API, personally. It junks things up and confuses users. If people feel keeping 1000 items in RAM is too risky, I'll just revert to putting this only in CMS.

But, I want to reiterate, I don't see how that can rationally be our position given that:
1. `Semigroup.sumOption` on tuples batches with size of 1000 already.
2. default sumByKey keeps 100K keys AND values.
3. the new TypedMapsideReduce *fully* materializes the full flatMap output into memory (no matter the size) before pushing over to the summing buffer.
4. probably many more cases of in-memory buffering of a constant number of items.

Are we really wanting to set a new policy of something like only 1 item of each type in memory? Or is it 100 would be okay, but now 1000? I'm not sure what people think we should ideally do in these cases.